packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,384 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	tokens,
+} from "@fluentui/react-components";
+import { DataEdit, FluidObjectValueNode, Primitive } from "@fluid-experimental/devtools-core";
+
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableValueView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Render data that is Editable with its corresponding UI
+ *
+ * @remarks {@link ContainerFeaturesContext} must be set in order to use this component.
+ */
+export function EditableView(props: EditableViewProps): React.ReactElement {
+	const { node, containerKey, label } = props;
+
+	/**
+	 * State to store whether the user is currently editing data
+	 */
+	const [editing, setEditing] = React.useState<boolean>(false);
```

Comment:
Nit: I usually recommend using strictly boolean verbs for boolean variables. E.g. `isEditing` instead of `editing`.